tox.ini

```diff
@@ -58,6 +61,14 @@ commands =
 whitelist_externals =
     find
 
+# generic environment that should cover anything that is not linter or a unit/functional test
+[testenv:check]
+# reuse existing envdir to avoid increased footprint
+envdir = {toxworkdir}/py27-ansible25-unit
+commands =
+    # verifies that pytest collection works without specific folders mentioned
+    python -m pytest --collect-only --html={envlogdir}/report.html --self-contained-html
```

Comment:
Testing that collection works and report generation is saved are things that you can do in your personal setup easily. I'd rather not add any further test environments especially something that is for a "generic" need (read: YAGNI).

Reply:
I have to disagree here with a clear example: we already have `metadata-validation` one which is not always run but which can also break the code.  My intention was to merge these two into a single "check" one.

That pattern does not scale well. The total number of tox environments needs to be kept under control, the same applies to travis jobs, and nobody says that they need to match 1:1.

I am more than willing to hear about which ideas you have.

AFAIK, YAGNI does not apply here, collection was broken and we need to avoid regression, especially as is very easy to break it.

Regarding python-html part, it is not as simple as you could imagine because while I can locally define PYTEST_ADDOPTS with extra params for it, running it from tox it will fail because pytest-html will not be installed into the venv managed by tox. Also, the report will not be stored in tox log specific to each environment because I would not be able to use `{envlogdir}` expansion. So, shortly it would be impossible to use it without altering the source-code. 

Sadly travis does not provide any support for accesing build artifacts, as we would have to upload them ourselves to an external storage. With upcoming zuul situation is much better as it does also knows to collect tox logs dirs by default. If you are curious click on openstack-tox-molecule job from https://review.opendev.org/#/c/667056/ and you can see the report.

Reply:
We should use a test coverage service instead of pytest-html, no? Why not https://coveralls.io/ or something similar?

Reply:
I am OK with "check" env, we can afford a small fast one, no? Maybe it's just easier to do it this way.